src/aof.c

```diff
@@ -1643,8 +1646,12 @@ int loadAppendOnlyFiles(aofManifest *am) {
     server.aof_rewrite_base_size = server.aof_current_size;
     server.aof_fsync_offset = server.aof_current_size;
 
+    /* If we succeeded to load one of the AOF files, and the last AOF file is empty,
+     * the result is AOF_OK (and not AOF_EMPTY) */
+    if (aof_loaded && ret == AOF_EMPTY) ret = AOF_OK;
+
 cleanup:
-    stopLoading(ret == AOF_OK);
+    stopLoading(ret == AOF_OK || ret == AOF_TRUNCATED || ret == AOF_EMPTY);
```

Comment:
already outdated, right?
```c
    /* Here we calculate the total size of all BASE and INCR files in
     * advance, it will be set to `server.loading_total_bytes`. */
    total_size = getBaseAndIncrAppendOnlyFilesSize(am, &status);
    if (status != AOF_OK) {
        /* If an AOF exists in the manifest but not on the disk, we consider this to be a fatal error. */
        if (status == AOF_NOT_EXIST) status = AOF_FAILED;

        return status;
    } else if (total_size == 0) {
        return AOF_EMPTY;
    }
```

Reply:
Yes, it seems ok.